_posts/2022-05-19-theory-of-mind.md

```diff
@@ -0,0 +1,144 @@
+
+## Implications for Goal-Oriented Chatbots
+
+Open-domain chat has one important goal, _engagement with the user_. The user engages with the chatbot _if the chatbot is entertaining the user_. For this statement to hold true, the bot should appropriately make responses, which in-turn requires modeling the user, i.e., having a Theory of Mind. The _degree of engagement_ can be seen as a measure of _degree of ToM_ of the chatbot.
+
+Testing ToM is straight-forward for Open-domain (chit-chat) chatbot. However, this is tricky for goal-oriented chatbots, as they are designed to handle dialog under a specific domain. False-belief task defined on one domain maybe out-of-domain for another domain.
+
+Open-domain dialog is a strict generalization of goal-oriented dialog. However, goal-oriented may have goals which are defined differently from _engagement_. In many call-center settings, _call resolution_ is the most important goal. However, when voice chatbots are used in-place of human agents in call centers, a new and different behaviour of users arises: _call drop_. Users simply drop from the call if they:
+1. get frustrated (due to mishearing, poor reasoning capabilities etc.).
+2. think the chatbot is _incapable_ of answering their queries, even if the bot is capable. This is a false-belief of the user, and the chatbot is unable to correct the user's false-belief.
+Call drops occur in a major chunk of the calls (40-50%).
+
+Most chatbots in the industry are designed in a way that assumes _the user trusts the chatbot and has infinite patience_. Chatbot's behaviour is apparrently designed to optimize for resolving queries of the user, however, not to _inspire trust in the user that the bot is capable to resolve queries_.
+
+There are two possible ways to "solve" this problem:
+1. `Explicit`: Design the product in a way that inspires trust. Come up with the _best possible_ responses for all possible combination of dialog history and user states.
+2. `Implicit`: Design the product in a top-down fashion rather than a bottom-up. Many believe that optimizing components with their local objective (Word-error-rate for ASR, F1 Scores for Intent classifiers etc.) would lead to a higher resolution rate. In biological systems, the higher-order function (survival) dictates lower order function (communication, language). Learning to communicate better can not ensure survival on its own. However, learning to survive _may lead to a better ability to communicate_. In other words, optimize ML models against objective (resolution rate) in-addition to the local objective. This _will_ force the chatbot to behave in a way that inspires trust from the user and effectively learn to have theory of mind of the users.
+
+Clearly, the first method doesn't seem to be working well, and the second method has the clear advantage of being data-driven and scalable.
```

Comment:
A person not intricately working in this domain, might not be able to come to this conclusion "clearly".

Reply:
Agreed, but I've pointed out that the first method is how the industry does things right now, and how bad it is.